### PR TITLE
(Feat) add fishsoop dest_path

### PR DESCRIPTION
**Pull request by lbesnard**

### CODE REVIEW

**Comment by lbesnard** [comment]
To be merged once everything is confirmed by facility

File: aodndata/soop/soop_fishsoop.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import matplotlib.pyplot as plt
+from aodncore.pipeline import HandlerBase, PipelineFile, PipelineFilePublishType
+from aodncore.util.misc import get_pattern_subgroups_from_string
+from netCDF4 import Dataset
+from numpy import ma
+
+FISHSOOP_FILE_PATTERN = re.compile(r"""
+                              IMOS_SOOP-FishSOOP_T_
+                              (?P<nc_time_cov_start>[0-9]{8}T[0-9]{6}Z)_
```
Comment (bpasquer) [inline aodndata/soop/soop_fishsoop.py:15]: The data code should be TP

File: test_aodndata/soop/test_fishsoop.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from aodncore.pipeline.exceptions import InvalidFileNameError
+from aodndata.soop.soop_fishsoop import dest_path_soop_fishsoop
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+
+class TestSoopFishSoop(unittest.TestCase):
+
+    def test_dest_path_soop_fishsoop(self):
+        good_nc_nrt = os.path.join(TEST_ROOT, 'IMOS_SOOP-FishSOOP_T_20240310T094642Z_FV00_786.nc')
```
Comment (bpasquer) [inline test_aodndata/soop/test_fishsoop.py:13]: see comment above

File: test_aodndata/soop/test_fishsoop.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        self.assertEqual(dest_path_soop_fishsoop(good_nc_nrt),
+                         'IMOS/SOOP/SOOP-FishSoop/REALTIME/2024/03/{basename}'.format(
+                             basename=os.path.basename(good_nc_nrt)),
+        )
+
+        good_nc_dm = os.path.join(TEST_ROOT, 'IMOS_SOOP-FishSOOP_T_20240310T094642Z_FV01_786.nc')
+        self.assertEqual(dest_path_soop_fishsoop(good_nc_dm),
```
Comment (bpasquer) [inline test_aodndata/soop/test_fishsoop.py:20]: same as above

**Review by bpasquer — CHANGES_REQUESTED** [review]

File: test_aodndata/soop/test_fishsoop.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+
+class TestSoopFishSoop(unittest.TestCase):
+
+    def test_dest_path_soop_fishsoop(self):
+        good_nc_nrt = os.path.join(TEST_ROOT, 'IMOS_SOOP-FishSOOP_T_20240310T094642Z_FV00_786.nc')
+        self.assertEqual(dest_path_soop_fishsoop(good_nc_nrt),
+                         'IMOS/SOOP/SOOP-FishSoop/REALTIME/2024/03/{basename}'.format(
```
Comment (bpasquer) [inline test_aodndata/soop/test_fishsoop.py:15]: FishSOOP

File: aodndata/soop/soop_fishsoop.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        serial_number = fields['serial_number']
+
+        if qc_level == 'FV00':
+            data_mode = 'REALTIME'
+        elif qc_level == 'FV01':
+            data_mode = "DELAYED"
+        else:
+            return ValueError(f"{qc_level} not matching expected qc_level")
+
+        return os.path.join('IMOS', 'SOOP', 'SOOP-FishSoop', data_mode,
```
Comment (bpasquer) [inline aodndata/soop/soop_fishsoop.py:38]: FishSOOP

File: test_aodndata/soop/test_fishsoop.py (hunk 4 above)
Comment (bpasquer) [reply test_aodndata/soop/test_fishsoop.py:15]: more SOOP

File: test_aodndata/soop/test_fishsoop.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+from aodncore.pipeline.exceptions import InvalidFileNameError
+from aodndata.soop.soop_fishsoop import dest_path_soop_fishsoop
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+
+class TestSoopFishSoop(unittest.TestCase):
+
+    def test_dest_path_soop_fishsoop(self):
+        good_nc_nrt = os.path.join(TEST_ROOT, 'IMOS_SOOP-FishSOOP_PT_20240310T094642Z_FV00_786.nc')
```
Comment (bpasquer) [inline test_aodndata/soop/test_fishsoop.py:13]: TP instead of PT

**Pull request merged**